src/ring_theory/int/basic.lean

```diff
@@ -341,3 +341,54 @@ instance decidable_int : decidable_rel (λ a b : ℤ, (multiplicity a b).dom) :=
 λ a b, decidable_of_iff _ finite_int_iff.symm
 
 end multiplicity
+
+lemma induction_on_primes {P : nat → Prop} (h₀ : P 0) (h₁ : P 1)
+  (h : ∀ p a : ℕ, nat.prime p → P a → P (p * a)) (n : ℕ) : P n :=
+begin
+  apply unique_factorization_monoid.induction_on_prime,
+  exact h₀,
+  { intros n h,
+    rw nat.is_unit_iff.1 h,
+    exact h₁, },
+  { intros a p _ hp ha,
+    exact h p a (nat.prime_iff_prime.2 hp) ha, },
+end
+
+lemma int.associated_nat_abs (k : ℤ) : associated k k.nat_abs :=
+associated_of_dvd_dvd (int.coe_nat_dvd_right.mpr (dvd_refl _)) (int.nat_abs_dvd.mpr (dvd_refl _))
+
+lemma int.prime_iff_nat_abs_prime {k : ℤ} : prime k ↔ nat.prime k.nat_abs :=
+begin
+  rw nat.prime_iff_prime_int,
+  rw prime_iff_of_associated (int.associated_nat_abs k),
+end
+
+lemma int.nat_abs_eq_nat_abs_iff {a b : ℤ} : a.nat_abs = b.nat_abs ↔ a = b ∨ a = -b :=
```

Comment:
This one should probable go somewhere closer to `int.nat_abs_eq`, though.

Reply:
I moved it to `init.data.int.basic` (which forced me to update the proof and I even got it shorter !)

Reply:
Are you sure you mean `init`? That's not part of mathlib.

Reply:
Actually, `int.nat_abs_eq` is in `init.int.basic`, which doesn't belong to mathlib but to lean.

Should I PR lean for this lemma ?

Reply:
I don't think that's a can of worms you want to open. If you did that, you'd have to wait for the next lean release before continuing with this PR. Perhaps @gebner can give better advice, since they modify Lean itself quite frequently.

Reply:
The general policy is that lemmas that are not used in core can and should go into mathlib.  So it's fine here.

Reply:
Ok, I guess this resolves this, thanks @gebner

Reply:
Oh, I didn't realize that `int.nat_abs_eq` was in core. Perhaps it could go in https://leanprover-community.github.io/mathlib_docs/data/int/basic.html#nat-abs though, since it doesn't depend on any `ring_theory`? How do people feel about that?

Reply:
I just moved it where you pointed, thanks :)